controllers/storageautoscaler/util.go

```diff
@@ -0,0 +1,109 @@
+package storageautoscaler
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	prometheusconfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/platform"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getPrometheusURL(operatorNamespace string) (string, error) {
+	prometheusURL := "https://prometheus-k8s.openshift-monitoring.svc.cluster.local:9091"
+	if isROSAHCP, err := platform.IsPlatformROSAHCP(); err != nil {
+		return "", err
+	} else if isROSAHCP {
+		prometheusURL = fmt.Sprintf("https://prometheus.%s.svc.cluster.local:9339", operatorNamespace)
+	}
+
+	return prometheusURL, nil
+}
+
+func getScrpaer(ctx context.Context, operatorNamespace string, log logr.Logger) (prometheusv1.API, error) {
+	prometheusURL, err := getPrometheusURL(operatorNamespace)
+	if err != nil {
+		log.Error(err, "Failed to determine if ROSA HCP cluster")
+		return nil, err
+	}
+
+	caCertPath := "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
```

Comment:
Define `/var/run/secrets/kubernetes.io/serviceaccount` as a separate var, then join it with the filenames, so you don't have to duplicate the path